File: tests/fixtures/ruby/mixed_priority.rb
# FIXTURE: Ruby file with mixed priority items
# TESTS: Truncation priority testing

require 'json'
require 'net/http'
require 'uri'

Config = Struct.new(:host, :port, :timeout)

module LogLevel
  DEBUG = 0
  INFO = 1
  WARN = 2
  ERROR = 3
end

class Server
  attr_reader :config

  def initialize(config)
    @config = config
  end

  def start
    puts "Starting on #{config.host}:#{config.port}"
  end

  def stop
    puts "Stopping server"
  end
end

def process_request(url, config)
  return -1 if url.nil? || config.nil?
  puts "Processing: #{url} on port #{config.port}"
  0
end

def validate_url(url)
  URI.parse(url)
  true
rescue URI::InvalidURIError
  false
end

MAX_RETRIES = 3
DEFAULT_PORT = 8080

File: tests/fixtures/csharp/mixed_priority.cs
// FIXTURE: C# file with mixed priority items
// TESTS: Truncation priority testing

using System;
using System.Collections.Generic;

namespace MyApp
{
    public enum LogLevel
    {
        Debug,
        Info,
        Warn,
        Error
    }

    public interface IConfigService
    {
        string GetHost();
        int GetPort();
    }

    public class Config
    {
        public string Host { get; set; }
        public int Port { get; set; }
        public int Timeout { get; set; }
    }

    public class Server
    {
        private readonly Config _config;

        public Server(Config config)
        {
            _config = config;
        }

        public void Start()
        {
            Console.WriteLine($"Starting on {_config.Host}:{_config.Port}");
        }

        public void Stop()
        {
            Console.WriteLine("Stopping server");
        }
    }

    public static int ProcessRequest(string url, Config config)
    {
        if (string.IsNullOrEmpty(url) || config == null)
            return -1;
        Console.WriteLine($"Processing: {url} on port {config.Port}");
        return 0;
    }
}
